apps/sms/test/marionette/notification_test.js

```diff
@@ -228,4 +229,57 @@ marionette('Message notification tests', function() {
       assertMessagesIsInCorrectState();
     });
   });
+
+  suite('Display a conversation from a notification', function() {
+    var messagesStorage;
+    var thread1, thread2;
+
+    setup(function() {
+      thread1 = ThreadGenerator.generate({
+        numberOfMessages: 500,
+        participants: ['999']
+      });
+      thread2 = ThreadGenerator.generate({
+        numberOfMessages: 5,
+        participants: ['888']
+      });
+
+      messagesStorage = [ thread1, thread2 ];
+    });
+
+    test('Clicking a notification while a conversation is loading', function() {
+      messagesApp.launch();
+      storage.setMessagesStorage(messagesStorage);
+
+      var inbox = new InboxView(client);
+      var conversation = inbox.goToConversation(thread1.id);
+
+      // Receive 'notification' system message and fill up the storage.
+      messagesApp.sendSystemMessage('notification', {
+        clicked: true,
+        data: { threadId: thread2.id, id: thread2.messages[0].id }
+      });
+
+      var thread2Ids = thread2.messages.map(
+        function(message) { return message.id; }
+      );
+
+      // Our testing strategy is:
+      // 1. Wait that all messages are rendered for thread2.
+      thread2Ids.forEach(function(messageId) {
+        conversation.findMessage(messageId);
+      });
+
+      // 2. Look at all rendered messages and see if one of these is part of
+      // thread1.
+      conversation.messages(
```

Comment:
__optional nit:__ we can probable fail earlier with additional assert, but it's veeeery optional :)
```
var messages = conversation.messages({ first: thread2Ids.length + 1 });

assert.equal(messages.length, thread2Ids.length);

messages.forEach(.....)
```